lib/portlayer/exec/container.go

```diff
@@ -151,6 +151,20 @@ func (s State) String() string {
 	return ""
 }
 
+// CurrentState returns current state.
+func (c *Container) CurrentState() State {
+	c.m.Lock()
+	defer c.m.Unlock()
+	return c.state
+}
+
+// SetState changes container state.
+func (c *Container) SetState(s State) {
```

Comment:
unless we've untangled the state the incoming events can change the container state, but that should not reflect in the handle.